Assets/MixedRealityToolkit/Utilities/MRTKServicesUtils.cs

```diff
@@ -0,0 +1,110 @@
+        /// The active instance of the input system
+        /// </summary>
+        public static IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+
+        private static IMixedRealityBoundarySystem boundarySystem = null;
+
+        /// <summary>
+        /// The active instance of the boundary system
+        /// </summary>
+        public static IMixedRealityBoundarySystem BoundarySystem
+        {
+            get
+            {
+                if (boundarySystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityBoundarySystem>(out boundarySystem);
+                }
+                return boundarySystem;
+            }
+        }
+
+        private static IMixedRealitySpatialAwarenessSystem spatialAwarenessSystem = null;
+
+        /// <summary>
+        /// The active instance of the spatial awareness system
+        /// </summary>
+        public static IMixedRealitySpatialAwarenessSystem SpatialAwarenessSystem
+        {
+            get
+            {
+                if (spatialAwarenessSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealitySpatialAwarenessSystem>(out spatialAwarenessSystem);
+                }
+                return spatialAwarenessSystem;
+            }
+        }
+
+        private static IMixedRealityTeleportSystem teleportSystem = null;
+        /// <summary>
+        /// The active instance of the teleport system
+        /// </summary>
+        public static IMixedRealityTeleportSystem TeleportSystem
+        {
+            get
+            {
+                if (teleportSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityTeleportSystem>(out teleportSystem);
+                }
+
+                return teleportSystem;
+            }
+        }
+
+        private static IMixedRealityDiagnosticsSystem diagnosticsSystem = null;
+        /// <summary>
+        /// The active instance of the diagnostics system
+        /// </summary>
+        public static IMixedRealityDiagnosticsSystem DiagnosticsSystem
+        {
+            get
+            {
+                if (diagnosticsSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityDiagnosticsSystem>(out diagnosticsSystem);
+                }
+                return diagnosticsSystem;
+            }
+        }
+
+        public static void InvalidateReferences()
+        {
+            diagnosticsSystem = null;
+            teleportSystem = null;
+            spatialAwarenessSystem = null;
+            boundarySystem = null;
+            inputSystem = null;
```

Comment:
We're missing references to the scene system and camera system core services.

Reply:
oh didn't even know these were available now, I just searched MRTK for the registry usage. Will add!